unlock-app/src/__tests__/utils/signature.test.js

```diff
@@ -0,0 +1,58 @@
+import generateJWTToken from '../../utils/signature'
+import Web3Service from '../../services/web3Service'
+import configure from '../../config'
+
+var nock = require('nock')
+
+var nockDefs = nock.loadDefs(__dirname + '/fixtures/signatureFixture.json')
+nock.define(nockDefs)
+
+const defaultState = {
+  network: {
+    name: 'test',
+  },
+  provider: 'HTTP',
+  account: {},
+}
+
+const { providers } = configure(global)
+
+describe('generateJWTToken', () => {
+  describe('when authorized to sign the payload', () => {
+    it('generates a JWT signed by the address holder', done => {
+      Date.now = jest.fn(() => 1546219627000)
+      let web3Service = new Web3Service(providers)
+
+      web3Service.connect(Object.assign({}, defaultState)).then(() => {
+        generateJWTToken(
+          web3Service,
+          '0xaaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2',
+          { data: 'ddd' }
+        ).then(result => {
+          expect(result).toBe(
+            'eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJkYXRhIjoiZGRkIiwiaWF0IjoxNTQ2MjE5NjI3LCJleHAiOjE1NDYyMTk2MzAsImlzcyI6IjB4YWFhZGVlZDRjMGI4NjFjYjM2ZjRjZTAwNmE5YzkwYmEyZTQzZmRjMiJ9.0xd64fd442c30596b2861f60d8e55207aa239df32303689463ea4f5ab48ee9c4992eb6db40d8f8c5b568413f3963edfdb708ed788369c87fc93b1aca95222e54e401'
```

Comment:
Would be good to extract large values and pair them into an { address, jwt } struct for easier reading